src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FallThroughCheck.java

```diff
@@ -152,35 +112,76 @@
  * Example:
  * </p>
  * <pre>
- * switch (i) {
- *   case 1:
- *     break;
- *   case 2: // Previous case: OK
- *           // This case: violation, last case must have statement that transfer control
- *     i++;
+ * class Example2 {
+ *   public int foo() {
+ *     int i = 0;
+ *     while(i &gt;= 0) {
+ *       switch(i) {
+ *         case 1:
+ *           i++;  // falls-thru
```

Comment:
Do not add extra comments, all should be same in all examples.
Change should be in  location of `// violation`

Reply:
Done